### PR TITLE
[libevent] Update the patch and enable uwp support.

### PR DESCRIPTION
**Pull request by lemourin**
Tagged release is +2 years old, fix-android-threads patch got upstreamed so it is no longer necessary here. Also, includes a small patch which makes it build on uwp.

### CODE REVIEW

File: ports/libevent/vcpkg.json [hunk 1]
```diff
@@ -1,10 +1,9 @@
 {
   "name": "libevent",
   "version": "2.1.12",
-  "port-version": 7,
+  "port-version": 8,
   "description": "An event notification library",
   "homepage": "https://github.com/libevent/libevent",
```
Comment (MonicaLiu0311) [inline ports/libevent/vcpkg.json:6]:
```suggestion
  "homepage": "https://github.com/libevent/libevent",
  "license": "BSD-3-Clause",
```

Comment (MonicaLiu0311) [reply ports/libevent/vcpkg.json:6]: A `license` is required.

File: ports/libevent/vcpkg.json [hunk 2]
```diff
@@ -1,10 +1,9 @@
 {
   "name": "libevent",
   "version": "2.1.12",
```
Comment (dg0yt) [inline ports/libevent/vcpkg.json:3]: If the source code is no longer at `release-2.1.12-stable`, the version cannot remain at `2.1.12`. I would suggest to include the commit date as in `2.1.12+yyyymmdd`. Cf. port libssh.

**Comment by MonicaLiu0311** [comment]
A CI error, unrelated to this PR:
```
Installing 187/209 brpc:x64-linux...
Building brpc[core]:x64-linux...
-- Downloading https://github.com/apache/incubator-brpc/archive/29491107cbf405a494aaf80ee32344ba34e1d7e4.tar.gz -> apache-incubator-brpc-29491107cbf405a494aaf80ee32344ba34e1d7e4.tar.gz...
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Detected CI environment: Azure_Pipelines
[DEBUG] 1000: popen( curl --fail -L "[https://vcpkgassetcacheeastasia.blob.core.windows.net/cache/bd4c67967796592030903041ddb9205e24c9f196e63ebc153e08fbce723d93d27cd4f30f3c2cf904a93cda66ffa9db7d465d6e5fdac27a045ae84afad3dd1dc3?sp=racwdl&st=2022-10-02T19:45:04Z&se=2023-02-01T00:00:00Z&spr=https&sv=2021-06-08&sr=c&sig=EnwpGwYUcR9IlkPEfC2E6l84mOU69Joqoq2GqKpnm74%3D"](https://vcpkgassetcacheeastasia.blob.core.windows.net/cache/bd4c67967796592030903041ddb9205e24c9f196e63ebc153e08fbce723d93d27cd4f30f3c2cf904a93cda66ffa9db7d465d6e5fdac27a045ae84afad3dd1dc3?sp=racwdl&st=2022-10-02T19:45:04Z&se=2023-02-01T00:00:00Z&spr=https&sv=2021-06-08&sr=c&sig=EnwpGwYUcR9IlkPEfC2E6l84mOU69Joqoq2GqKpnm74%3D%22) --create-dirs --output /mnt/vcpkg-ci/downloads/apache-incubator-brpc-29491107cbf405a494aaf80ee32344ba34e1d7e4.tar.gz.3593685.part 2>&1)
[DEBUG] 1000: cmd_execute_and_stream_data() returned 22 after    28387 us
[DEBUG] 1001: popen( curl --fail -L https://github.com/apache/incubator-brpc/archive/29491107cbf405a494aaf80ee32344ba34e1d7e4.tar.gz --create-dirs --output /mnt/vcpkg-ci/downloads/apache-incubator-brpc-29491107cbf405a494aaf80ee32344ba34e1d7e4.tar.gz.3593685.part 2>&1)
[DEBUG] 1001: cmd_execute_and_stream_data() returned 0 after  2459299 us
error: Failed to download from mirror set
error: https://vcpkgassetcacheeastasia.blob.core.windows.net/cache/bd4c67967796592030903041ddb9205e24c9f196e63ebc153e08fbce723d93d27cd4f30f3c2cf904a93cda66ffa9db7d465d6e5fdac27a045ae84afad3dd1dc3?*** SECRET ***: curl failed to download with exit code 22
error: File does not have the expected hash:
url: https://github.com/apache/incubator-brpc/archive/29491107cbf405a494aaf80ee32344ba34e1d7e4.tar.gz
File: /mnt/vcpkg-ci/downloads/apache-incubator-brpc-29491107cbf405a494aaf80ee32344ba34e1d7e4.tar.gz.3593685.part
Expected hash: bd4c67967796592030903041ddb9205e24c9f196e63ebc153e08fbce723d93d27cd4f30f3c2cf904a93cda66ffa9db7d465d6e5fdac27a045ae84afad3dd1dc3
Actual hash: d484f56c84d6a08dad22eab5c8cabe9f761382e5e938041b7db92ce9e5f493989f06793d55bebb70b2fa8ddcbf8e5e78ee393ea6957c244e961a899de7d3f014
[DEBUG] /mnt/vss/_work/1/s/src/vcpkg/base/downloads.cpp(956): 
[DEBUG] Time in subprocesses: 2487686 us
[DEBUG] Time in parsing JSON: 11 us
[DEBUG] Time in JSON reader: 0 us

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
      
      Failed to download file with error: 1
      If you use a proxy, please check your proxy setting. Possible causes are:
      
      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`./ This is not correct, because `[https://`](https://%60/) prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port`/ to both
         HTTP_PROXY and HTTPS_PROXY instead.
      
      2. You are using Fiddler. Currently a bug (***/issues/17752)
         will set HTTPS_PROXY to `https://fiddler_address:port`/ which lead to problem 1 above.
         Workaround is open Windows 10 Settings App, and search for Proxy Configuration page,
         Change `http=address:port;https=address:port` to `address`, and fill the port number.
      
      3. Your proxy's remote server is out of service.
      
      In future vcpkg releases, if you are using Windows, you no longer need to set
      HTTP(S)_PROXY environment variables. Vcpkg will simply apply Windows IE Proxy
      Settings set by your proxy software. See (***-tool/pull/49)
      and (***-tool/pull/77)
      
      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:273 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_github.cmake:106 (vcpkg_download_distfile)
  ports/brpc/portfile.cmake:1 (vcpkg_from_github)
  scripts/ports.cmake:147 (include)


error: building brpc:x64-linux failed with: BUILD_FAILED
Elapsed time to handle brpc:x64-linux: 3.252 s
```
Reproduced locally.

File: ports/libevent/fix-LibeventConfig_cmake_in_path.patch [hunk 3]
```diff
@@ -1,22 +1,27 @@
...
+                     PATHS "${_INSTALL_PREFIX}/lib"
+-                    NO_DEFAULT_PATH)
++                    )
+         find_library(_event_lib_rel
                      NAMES "event_${_comp}"
                      PATHS "${_INSTALL_PREFIX}/lib"
 -                    NO_DEFAULT_PATH)
 +                    )
-         if(_event_lib)
+         if(_event_lib_rel OR _event_lib_dbg)
```
Comment (dg0yt) [inline ports/libevent/fix-LibeventConfig_cmake_in_path.patch:25]: This change extends a pattern which I find questionable: The remove of `NO_DEFAULT_PATH`. Maybe this was a misguided attempt to deal with `NAMES "event_${_comp}"` not working for the debug build? Then the option should be brought back into both calls.

In addition, after this change, `_event_lib_rel` and `_event_lib_dbg` are set, and `_event_lib` isnt't. But I fail to spot a change to the variable which is used. Is it already handled in existing code?

**Comment by MonicaLiu0311** [comment]
Wait for a short while, looking forward to the reply from upstream: https://github.com/libevent/libevent/issues/1410.

**Comment by MonicaLiu0311** [comment]
All features are tested successfully in the following triplet:
x86-windows
x64-windows
x64-windows-static

File: ports/libevent/vcpkg.json [hunk 4]
```diff
@@ -1,10 +1,9 @@
 {
   "name": "libevent",
-  "version": "2.1.12",
-  "port-version": 7,
+  "version": "2.1.12+20230128",
   "description": "An event notification library",
   "homepage": "https://github.com/libevent/libevent",
-  "supports": "!uwp",
```
Comment (MonicaLiu0311) [inline ports/libevent/vcpkg.json:7]: Please delete the line corresponding to [ci.baseline.txt](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt#L384-L385).

**Review by MonicaLiu0311 — CHANGES_REQUESTED** [review]

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**